src/vs/workbench/parts/files/browser/views/explorerView.ts

```diff
@@ -630,15 +660,31 @@ export class ExplorerView extends CollapsibleView {
 			}
 		}
 
+		if (e.gotUpdated()) {
+			// Check updated only if sortOrder === MODIFIED
+			const config = this.configurationService.getConfiguration<IFilesConfiguration>();
+			if (config && config.explorer && config.explorer.sortOrder === SortOrderConfiguration.MODIFIED) {
```

Comment:
Why do you need to resolve the config here again? You should be able to just use `this.sortOrder`?